src/Features/Core/Portable/Completion/CompletionService.cs

```diff
@@ -110,12 +111,59 @@ public virtual TextSpan GetDefaultItemSpan(SourceText text, int caretPosition)
         /// This value is null when the commit was caused by the [TAB] or [ENTER] keys.</param>
         /// <param name="cancellationToken"></param>
         public virtual Task<CompletionChange> GetChangeAsync(
-            Document document, 
-            CompletionItem item, 
-            char? commitCharacter = null, 
+            Document document,
+            CompletionItem item,
+            char? commitCharacter = null,
             CancellationToken cancellationToken = default(CancellationToken))
         {
             return Task.FromResult(CompletionChange.Create(ImmutableArray.Create(new TextChange(item.Span, item.DisplayText))));
         }
+
+        /// <summary>
+        /// Given a list of completion items that match the current code typed by the user,
+        /// returns the item that is considered the best match, and whether or not that
+        /// item should be selected or not.
+        /// 
+        /// itemToFilterText provides the values that each individual completion item should
+        /// be filtered against.
+        /// </summary>
+        internal virtual ImmutableArray<CompletionItem> ChooseBestItems(
+            Document document,
+            ImmutableArray<CompletionItem> filteredItems,
+            IReadOnlyDictionary<CompletionItem, string> itemToFilterText)
```

Comment:
Yes.  That's exactly what i said in the PR message :)  I'm looking for ideas on how to make this better.  I have two other options i can think of.  But i welcome other thoughts you may have on this.